fuse_models/include/fuse_models/common/sensor_proc.h

```diff
@@ -218,14 +218,27 @@ inline void validatePartialMeasurement(
  * @param[in] tf_buffer - The transform buffer with which we will lookup the required transform
  * @param[in] input - The message to transform. Source frame and stamp are dictated by its header.
  * @param[in,out] output - The transformed message. Target frame is dictated by its header.
+ * @param [in] timeout - Optional. The maximum time to wait for a transform to become available.
  * @return true if the transform succeeded, false otherwise
  */
 template <typename T>
-bool transformMessage(const tf2_ros::Buffer& tf_buffer, const T& input, T& output)
+bool transformMessage(
+  const tf2_ros::Buffer& tf_buffer,
+  const T& input,
+  T& output,
+  const ros::Duration& tf_timeout = ros::Duration(0, 0))
 {
   try
   {
-    const auto trans = tf_buffer.lookupTransform(output.header.frame_id, input.header.frame_id, input.header.stamp);
+    auto trans = geometry_msgs::TransformStamped();
+    if (tf_timeout.isZero())
+    {
+      trans = tf_buffer.lookupTransform(output.header.frame_id, input.header.frame_id, input.header.stamp);
```

Comment:
I'm too lazy to look, but does this call just end up calling the other overload with a duration of 0 anyway? What happens if you always just pass `tf_timeout`, even if it's 0?

Reply:
It does not. The `lookupTransform()` method with the timeout calls the `canTransform()` version with a timeout internally, before calling the base class `lookupTransform()` method without the timeout. 
https://github.com/ros/geometry2/blob/noetic-devel/tf2_ros/src/buffer.cpp#L119

Whereas the non-timeout version of `lookupTransform()` _is_ the base class version.

There seemed like enough unneeded operations in the `canTransform()` implementation to make this additional bit of ugliness worthwhile.